src/Tests/ExecutionPathTracerTests.cs

```diff
@@ -67,11 +67,12 @@ private void PruneNonDeterministicGates(IEnumerable<Operation>? operations)
         }
 
         // Helper functions for tests
-        public Operation ControlledX(int[] controlId, int targetId) =>
+        public Operation ControlledX(int[] controlId, int targetId, bool isAdjoint = false) =>
```

Comment:
Thanks for fixing this! I'm slightly surprised that the execution path tracer is sensitive to the `Controlled X` operation being adjointed, though; seems like a reasonable thing to fix going forward after this is merged in, perhaps?